paseos/activities/activity_manager.py

```diff
@@ -0,0 +1,144 @@
+            power_consumption_in_watt=power_consumption_in_watt,
+            on_termination_function=on_termination_function,
+            constraint_function=constraint_function,
+            _dynamic=False,
+        )
+
+        logger.debug(f"Registered activity {self._activities[name]}")
+
+    def perform_activity(
+        self,
+        name: str,
+        activity_func_args: list = None,
+        termination_func_args: list = None,
+        constraint_func_args: list = None,
+    ):
+        """Perform the specified activity. Will advance the simulation if automatic clock is not disabled.
+
+        Args:
+            name (str): Name of the activity to perform.
+            activity_func_args (list, optional): Arguments for the activity function. Defaults to None.
+            termination_func_args (list, optional): Arguments for the termination function. Defaults to None.
+            constraint_func_args (list, optional): Arguments for the constraint function. Defaults to None.
+        """
+        # Check if activity exists and if it already had consumption specified
+        assert (
+            name in self._activities.keys()
+        ), f"Activity not found. Declared activities are {self._activities.keys()}"
+        activity = self._activities[name]
+        logger.debug(f"Performing activity {activity}")
+
+        assert (
+            activity.power_consumption_in_watt >= 0
+        ), "Power consumption has to be positive but was specified as " + str(
+            activity.power_consumption_in_watt
+        )
+
+        activity_runner = ActivityRunner(
+            name=name,
+            activity_func=activity.activity_function,
+            constraint_func=activity.constraint_function,
+            termination_func=activity.on_termination_function,
+            termination_args=termination_func_args,
+            constraint_args=constraint_func_args,
+        )
+
+        async def job():
+            processor = ActivityProcessor(
+                update_interval=self._paseos_update_interval,
+                power_consumption_in_watt=activity.power_consumption_in_watt,
+                paseos_instance=self._paseos_instance,
+                activity_runner=activity_runner,
+                advance_paseos_clock=self._paseos_instance.use_automatic_clock,
+            )
+
+            await asyncio.wait(
+                [
+                    processor.start(),  # noqa: F821
+                    activity_runner.start(activity_func_args),
+                ],
+                return_when=asyncio.ALL_COMPLETED,
+            )
+            self._paseos_instance._is_running_activity = False
+            del processor
+
+        # Workaround to avoid error when executed in a Jupyter notebook.
+        if is_notebook():
+            return job()
+        else:
+            # Run activity and processor
+            asyncio.run(job())
+
+        logger.info(f"Activity {activity} completed.")
```

Comment:
Do we have a preferred way of getting an output from the activity? The tests use a C-like approach by writing into the provided argument but it is not very pythonic.

Reply:
I updated #42 to mention this